test/extensions/filters/http/ext_authz/ext_authz_test.cc

```diff
@@ -2133,6 +2133,11 @@ TEST_F(HttpFilterTest, EmitDynamicMetadata) {
       .WillOnce(Invoke([&response](const std::string& ns,
                                    const ProtobufWkt::Struct& returned_dynamic_metadata) {
         EXPECT_EQ(ns, "envoy.filters.http.ext_authz");
+        // Check timing metadata correctness
+        EXPECT_TRUE(returned_dynamic_metadata.fields().at("ext_authz_duration").has_number_value());
+        (*response.dynamic_metadata.mutable_fields())["ext_authz_duration"] =
+            returned_dynamic_metadata.fields().at("ext_authz_duration");
+
```

Comment:
I'd like to see one of tests (probably one for the http filter and one for the network filter) check that the duration is an expected value. I think you could configure the  connection returned by `filter_callbacks_` in the test class to return an `SimulatedTimeSystem` (it may already) and use that to manipulate the result of the calls to `now` and then check that the duration is the expected value.

Reply:
I'll need to think a little bit on how to do this.